COVIDWatch iOS/Bluetooth/BluetoothPermission.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  BluetoothPermission.swift
+//  COVIDWatch iOS
+//
+//  Created by Madhava Jay on 13/4/20.
+//  Copyright © 2020 IZE. All rights reserved.
+//
+
+import Foundation
+import CoreBluetooth
+
+typealias BluetoothPermissionCallback = () -> Void
+
+class BluetoothPermission: NSObject, CBCentralManagerDelegate {
+    static let sharedInstance = BluetoothPermission()
+    override private init() {}
+    var centralManager: CBCentralManager?
+    var allowed: BluetoothPermissionCallback?
+    var denied: BluetoothPermissionCallback?
+
+    func checkPermission(_ allowed: @escaping () -> Void, _ denied: @escaping () -> Void) {
+        self.centralManager = CBCentralManager()
+        self.allowed = allowed
+        self.denied = denied
+        self.centralManager?.delegate = self
+    }
+
+    var isAuthorized: Bool {
+        if #available(iOS 13.0, *) {
+            return self.centralManager?.authorization == .allowedAlways
```

Comment:
fyi only - They added a static method for this in iOS 13.1 and this property was deprecated.

<img width="604" alt="Screen Shot 2020-04-13 at 7 40 26 AM" src="https://user-images.githubusercontent.com/2142301/79118197-2d13f500-7d5b-11ea-834e-19aab6316c16.png">

Reply:
Yeah I saw that, but i guess its only deprecated so still valid for now. Seems like a pain to deal with both and test for it right now so I just went with that. At least Xcode is good about stopping compilation when it's a problem so I guess this will be valid for quite a while longer.

Reply:
Seems to work so I guess its still dispatching to the class var?